src/modules/Client/Service.php

```diff
@@ -680,4 +681,51 @@ public function exportCSV(array $headers)
 
         return $this->di['table_export_csv']('client', 'clients.csv', $headers);
     }
+    /**
+     * Confirm password reset action.
+     *
+     * @return bool
+     *
+     * @throws \Box_Exception
+     */
+    public function pwreset_valid($data)
+    {
+        $required = [
+            'hash' => 'Hash required',
+        ];
+        $this->di['events_manager']->fire(['event' => 'onBeforePasswordResetClient']);
+        $this->di['validator']->checkRequiredParamsForArray($required, $data);
+
+        $reset = $this->di['db']->findOne('ClientPasswordReset', 'hash = ?', [$data['hash']]);
+        if (!$reset instanceof \Model_ClientPasswordReset) {
+            throw new \Box_Exception('The link have expired or you have already confirmed password reset.');
```

Comment:
You fixed the grammatical error in this exception elsewhere, please use the fixed version here